templates/plugin/monitoring_icinga2.yml.erb

```diff
@@ -0,0 +1,13 @@
+---
+:enabled: <%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::enabled') %>
+:server: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::server') %>"
+:api_cacert: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_cacert') %>"
+:api_user: "<%= scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_user') %>"
+<% api_password = scope.lookupvar('::foreman_proxy::plugin::monitoring::icinga2::api_password') -%>
+<% if (api_password.nil? && !api_password.empty?) || api_password == :undef %>
```

Comment:
This is raising an error in tests when api_password is nil. I'd suggest: `if api_password.nil? || api_password == :undef || api_password.empty?`